sonarcloud/resource_project_main_branch.go

```diff
@@ -0,0 +1,203 @@
+		return
+	}
+
+	request := project_branches.RenameRequest{
+		Project: plan.ProjectKey.Value,
+		Name:    plan.Name.Value,
+	}
+
+	err := r.p.client.ProjectBranches.Rename(request)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Could not create the main project branch",
+			fmt.Sprintf("The Rename request returned an error: %+v", err),
+		)
+		return
+	}
+}
+
+func (r resourceProjectMainBranch) Read(ctx context.Context, req tfsdk.ReadResourceRequest, resp *tfsdk.ReadResourceResponse) {
+	// Retrieve values from state
+	var state ProjectMainBranch
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Fill in api action struct
+	request := project_branches.ListRequest{
+		Project: state.ProjectKey.Value,
+	}
+
+	response, err := r.p.client.ProjectBranches.List(request)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Could not read the project branches",
+			fmt.Sprintf("The List request returned an error: %+v", err),
+		)
+		return
+	}
+
+	// Check if the main branch matches the declared main branch
+	if result, ok := findProjectMainBranch(response, state.Name.Value, state.ProjectKey.Value); ok {
+		diags = resp.State.Set(ctx, result)
+		resp.Diagnostics.Append(diags...)
+	} else {
+		resp.State.RemoveResource(ctx)
+	}
+}
+
+func (r resourceProjectMainBranch) Update(ctx context.Context, req tfsdk.UpdateResourceRequest, resp *tfsdk.UpdateResourceResponse) {
+	// Retrieve values from state
+	var state ProjectMainBranch
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Retrieve values from plan
+	var plan ProjectMainBranch
+	diags = req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	changed := changedAttrs(req, diags)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	projectKey := state.ProjectKey.Value
+	if _, ok := changed["project_key"]; ok {
+		projectKey = plan.ProjectKey.Value
+	}
+
+	name := state.Name.Value
+	if _, ok := changed["name"]; ok {
+		name = plan.Name.Value
+	}
+
+	request := project_branches.RenameRequest{
+		Project: projectKey,
+		Name:    name,
+	}
+
+	err := r.p.client.ProjectBranches.Rename(request)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Could not update the main project branch",
+			fmt.Sprintf("The Rename request returned an error: %+v", err),
+		)
+		return
+	}
+}
+
+func (r resourceProjectMainBranch) Delete(ctx context.Context, req tfsdk.DeleteResourceRequest, resp *tfsdk.DeleteResourceResponse) {
+	var state ProjectMainBranch
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// NOTE: according to docs, this may not work for main branches, and may require admin privilege
+	// https://github.com/reinoudk/go-sonarcloud/blob/main/sonarcloud/project_branches/project_branches_gen.go#L5
```

Comment:
☝️ Could/should `Delete` factor this into consideration and account for it in some way?

Reply:
Until #63 has been implemented it is not possible to run acceptance tests in CI, but if you create a test-organization as described [here](https://github.com/rewe-digital/terraform-provider-sonarcloud#developing-the-provider), you should be able to run the acceptance tests locally using `make testacc`.

As you put in the comment, `Delete` is only for non-main branches, so it cannot be used here. The NOOP would be possible (just remove it from state), but I'm wondering whether this resource should be a stand-alone resource. IMHO it would make more sense to have this as an attribute on the `sonarcloud_project` resource, i.e. `main_branch`.